rust/scx_utils/src/topology.rs

```diff
@@ -449,10 +463,24 @@ fn create_insert_cpu(cpu_id: usize, node: &mut Node, online_mask: &Cpumask) -> R
         span: Cpumask::new()?,
     });
 
+    let core_type = match avg_cpu_freq {
+        Some(avg_freq) => {
+            if max_freq > avg_freq {
+                CoreType::Big
+            } else if max_freq > 0 && max_freq < avg_freq {
+                CoreType::Little
+            } else {
+                CoreType::Big
+            }
```

Comment:
I think it could be a bit clearer the remove the last else and make the first if condition to `<=`. So the code should look like:

```
if max_freq >= avg_freq {
   // ...
}
} else if max_freq > 0 && max_freq < avg_freq {
  // ...
}
```